pkg/views/search-pagination-data.go

```diff
@@ -0,0 +1,6 @@
+package views
+
+type SearchPaginationData struct {
```

Comment:
It becomes a mess. You shouldn't split data between files required only for one webpage. And please name it in a way that describes the webpage in general. For example you have webpage that shows you a list of of jokes. So you could name a struct as `JokesPageParams` or something like this